machinist/mnode/templates.go

```diff
@@ -0,0 +1,116 @@
+package mnode
+
+import (
+	"bytes"
+	"errors"
+	"github.com/enfabrica/enkit/lib/logger"
+	"github.com/enfabrica/enkit/machinist/machinist_assets"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"text/template"
+)
+
+func InstallLibPam(l logger.Logger) error {
+	cmd := exec.Command("/bin/bash", "-c", string(machinist_assets.InstallLibPamScript))
+	o, err := cmd.Output()
+	l.Infof("output of installer is %s", string(o))
+	return err
+}
+
+func ReadSSHDContent(cafile, hostKey, hostCertificateFile string) ([]byte, error) {
+	tpl, err := template.New("ssh_server").Parse(machinist_assets.SSHDTemplate)
+	if err != nil {
+		return nil, err
+	}
+	type localConfig struct {
+		HostKeyFile         string
+		TrustedCAFile       string
+		HostCertificateFile string
+	}
+	l := localConfig{
+		TrustedCAFile:       cafile,
+		HostKeyFile:         hostKey,
+		HostCertificateFile: hostCertificateFile,
+	}
+	var r []byte
+	reader := bytes.NewBuffer(r)
+	err = tpl.Execute(reader, l)
+	return reader.Bytes(), err
+}
+
+// fetchLibNSSAutoUser will fetch the nss_autouser  lib that's embedded. The current build exports one .a and one .so.
+func fetchLibNSSAutoUser() ([]byte, error) {
+	for k, v := range machinist_assets.Data {
+		if strings.Contains(k, ".so") {
+			return v, nil
+		}
+	}
+	return nil, errors.New("the nssautouser has not been embedded properly")
+}
+
+// InstallNssAutoUser will attempt to install nssAutoUser if it can. Requires root. If not run as root or if not on
+// support operating system, it will error out.
+func InstallNssAutoUser(l logger.Logger) error {
+	nssBinary, err := fetchLibNSSAutoUser()
+	if err != nil {
+		return err
+	}
+	l.Infof("Successfully fetched the binary, installing now")
+	// TODO(adam): configure arm support for filepath writing and nss building
+	if err = ioutil.WriteFile("/lib/x86_64-linux-gnu/libnss_autouser.so.2", nssBinary, 0600); err != nil {
+		return err
+	}
+	out, err := ioutil.ReadFile("/etc/nsswitch.conf")
+	if err != nil {
+		return err
+	}
+	if !strings.Contains(string(out), "passwd:         files autouser systemd") {
```

Comment:
If you want to do this simply, you could run a .sh script or a simple regex replace with something like:

`passwd:(\s+)files(\s+.*)$` to replace with `passwd:\1files autouser\2`

Reply:
I'm going to do this later. It's changed from files to compat in ubuntu 20.04, and I feel like nsswitch should be set by the ansible script or provisioner.